ps_contactinfo.php

```diff
@@ -116,4 +117,59 @@ public function hookActionAdminStoresControllerUpdate_optionsAfter()
 
         return true;
     }
+
+    public function getContent()
+    {
+        $output = '';
+
+        if (Tools::isSubmit('submitContactInfo')) {
+            Configuration::updateValue('PS_CONTACT_INFO_DISPLAY_EMAIL', (int)Tools::getValue('PS_CONTACT_INFO_DISPLAY_EMAIL'));
+
+            foreach ($this->templates as $template) {
+                $this->_clearCache($template);
+            }
+
+            $output .= $this->displayConfirmation($this->trans('Settings updated.', array(), 'Admin.Notifications.Success'));
+
+            Tools::redirectAdmin($this->context->link->getAdminLink('AdminModules', true).'&conf=6&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name);
+        }
+
+        $helper = new HelperForm();
+        $helper->submit_action = 'submitContactInfo';
+
+        $field = array(
+            'type' => 'switch',
+            'label' => $this->trans('Display email address', array(), 'Admin.Actions'),
+            'name' => 'PS_CONTACT_INFO_DISPLAY_EMAIL',
+            'values' => array(
+                array(
+                    'id' => 'active_on',
+                    'value' => 1,
+                    'label' => $this->trans('Yes', array(), 'Admin.Global')
+                ),
+                array(
+                    'id' => 'active_off',
+                    'value' => 0,
+                    'label' => $this->trans('No', array(), 'Admin.Global')
+                )
+            )
+        );
+
+        $helper->fields_value['PS_CONTACT_INFO_DISPLAY_EMAIL'] = Configuration::get('PS_CONTACT_INFO_DISPLAY_EMAIL');
+
+        return $output.$helper->generateForm(array(
```

Comment:
```suggestion
        $output[] =$helper->generateForm(array(
```

And return imploded $output.
Easier to find where the output is appended

Reply:
smart ! 
(and done)